src/olympia/bandwagon/views.py

```diff
@@ -710,5 +710,17 @@ def get_object(self):
         return super(CollectionAddonViewSet, self).get_object()
 
     def get_queryset(self):
-        return CollectionAddon.objects.filter(
+        qs = CollectionAddon.objects.filter(
             collection=self.get_collection_viewset().get_object())
+        filter_param = self.request.GET.get('filter')
+        # We only filter list action.
+        include_deleted = (filter_param == 'with_hidden_and_deleted' or
+                           self.action != 'list')
+        # If deleted addons are requested, that implies hidden addons.
+        include_hidden = filter_param == 'with_hidden' or include_deleted
```

Comment:
Change the `include_*` variables to match the new filter names (see comment about the docs above)